packages/gasket-plugin-nextjs/generator/redux/redux/store.js

```diff
@@ -0,0 +1,17 @@
+const { configureMakeStore } = require('@gasket/redux');
+const { HYDRATE, createWrapper } = require('next-redux-wrapper');
+{{{reduxReducers.imports}}}
+
+// Basic hydrate reducer for next-redux-wrapper
+// @see: https://github.com/kirill-konshin/next-redux-wrapper#usage
+const rootReducer = (state, { type, payload }) => type === HYDRATE ? { ...state, ...payload } : state;
```

Comment:
This seems like a potential source of lots of head scratching. Someone dispatching to their store on server events might expect that state changes are going to be deeply, not shallowly merged and see existing state wiped out. Deep merges may also do unexpected things depending on the merge algorithm, so I get going the direction of having users define their own `rootReducer`.

It would be nice to raise some warnings somewhere in the documentation about this default `HYDRATE` handler since it's a bit counter-intuitive.

Reply:
👍🏻 Yea, I know I head scratched on this change for awhile. I'm thinking this doc is really better suited for the nextjs plugin, since the redux plugin doesn't and shouldn't be concerned with next or next-redux-wrapper. I'll move it over and add an upgrade section or another doc to describe this better.